end-to-end-tests/src/instance_launch.rs

```diff
@@ -132,7 +132,10 @@ async fn instance_launch() -> Result<()> {
         .split_once("-----BEGIN SSH HOST KEY KEYS-----")
         .and_then(|(_, s)| s.split_once("-----END SSH HOST KEY KEYS-----"))
         .and_then(|(lines, _)| {
-            lines.trim().lines().find(|line| line.starts_with("ssh-ed25519"))
+            lines
+                .trim()
+                .lines()
+                .find(|line| line.contains(public_key_str.clone().as_str()))
```

Comment:
This doesn't seem correct. This section of code is trying to extract the host's SSH key from the console output, so we can authenticate the host when we try to connect to it. That is not the same key as the one we created and sent to the instance to authenticate our user.

If we are failing to extract a host SSH key from the console output, either we aren't waiting long enough or something else is wrong.

Reply:
It was extracting a key, but the key it was extracting was wrong. I thought that for some reason there were multiple.